price_pusher/src/injective.ts

```diff
@@ -103,10 +122,126 @@ export class InjectivePriceListener implements PriceListener {
+    const chainGrpcAuthApi = new ChainGrpcAuthApi(this.grpcEndpoint);
+    const account = await chainGrpcAuthApi.fetchAccount(
+      this.injectiveAddress()
+    );
+    const { signBytes, txRaw } = createTransactionFromMsg({
+      sequence: account.baseAccount.sequence,
+      accountNumber: account.baseAccount.accountNumber,
+      message: msg,
+      chainId: "injective-888",
+      fee,
+      pubKey: this.wallet.toPublicKey().toBase64(),
+    });
+
+    const sig = await this.wallet.sign(Buffer.from(signBytes));
+
+    /** Append Signatures */
+    txRaw.setSignaturesList([sig]);
+
+    const txService = new TxGrpcClient(this.grpcEndpoint);
+    const txResponse = await txService.broadcast(txRaw);
+
+    return txResponse;
+  }
+
   async updatePriceFeed(
     priceIds: string[],
     pubTimesToPush: number[]
   ): Promise<void> {
-    console.log("dummy pushed");
+    if (priceIds.length === 0) {
+      return;
+    }
+
+    if (priceIds.length !== pubTimesToPush.length)
+      throw new Error("Invalid arguments");
+
+    let priceFeedUpdateObject;
+    try {
+      // get the latest VAAs for updatePriceFeed and then push them
+      priceFeedUpdateObject =
+        await this.cwPriceServiceConnection.getPriceFeedUpdateObject(priceIds);
+    } catch (e) {
+      console.error("Error fetching the latest vaas to push");
+      console.error(e);
+      return;
+    }
+
+    let updateFeeQueryResponse: UpdateFeeResponse;
+    try {
+      const api = new ChainGrpcWasmApi(this.grpcEndpoint);
+      const { data } = await api.fetchSmartContractState(
+        this.pythContract,
+        Buffer.from(
+          JSON.stringify({
+            get_update_fee: {
+              vaas: priceFeedUpdateObject.update_price_feeds.data,
+            },
+          })
+        ).toString("base64")
+      );
+
+      const json = Buffer.from(data as string, "base64").toString();
+      updateFeeQueryResponse = JSON.parse(json);
+    } catch (e) {
+      console.error("Error fetching update fee");
+      console.error(e);
+      return;
+    }
+
+    // TODO: add specific error messages
+    try {
+      const executeMsg = MsgExecuteContract.fromJSON({
+        sender: this.injectiveAddress(),
+        contractAddress: this.pythContract,
+        msg: priceFeedUpdateObject,
+        funds: [updateFeeQueryResponse],
+      });
+
+      const rs = await this.signAndBroadcastMsg(executeMsg);
+
+      if (rs.code !== 0) throw new Error("Error: transaction failed");
+
+      console.log("Succesfully broadcasted txHash:", rs.txHash);
+    } catch (e) {
+      console.error("Error executing messages");
+      console.log(e);
+    }
+  }
+}
+
+// FIXME: a better place for it while refactoring
+export class CwPriceServiceConnection extends PriceServiceConnection {
+  constructor(endpoint: string, config?: PriceServiceConnectionConfig) {
+    super(endpoint, config);
+  }
+
+  async getPriceFeedUpdateObject(priceIds: string[]): Promise<any> {
```

Comment:
If the only functionality is this we might just move it to the pusher 🤔

Reply:
Yeah. Actually, the version of pyth-common-js being used was an old one. getLatestVaas method was not public in the old one. 
Updated the version to latest. Now we can use getLatestVaas in this class too. 

Removed cwPriceServiceConnection in the latest commit.